src/compat/tracker.jl

```diff
@@ -1,105 +1,428 @@
 # end
 # @grad function _logabsdetjac_scale(a::TrackedMatrix, x::AbstractVector, ::Val{1})
 #     throw
 # end
 
 # implementations for Stacked bijector
 function logabsdetjac(b::Stacked, x::TrackedMatrix{<:Real})
-    return Tracker.collect([logabsdetjac(b, x[:, i]) for i = 1:size(x, 2)])
+    @views init = vcat(logabsdetjac(b, x[:, 1]))
+    return mapreduce(vcat, drop(eachcol(x), 1); init = init) do c
+        logabsdetjac(b, c)
+    end
 end
 # TODO: implement custom adjoint since we can exploit block-diagonal nature of `Stacked`
 function (sb::Stacked)(x::TrackedMatrix{<:Real})
-    init = reshape(sb(x[:, 1]), :, 1)
-    return Tracker.collect(mapreduce(i -> sb(x[:, i]), hcat, 2:size(x, 2); init = init))
-end
\ No newline at end of file
+    @views init = reshape(sb(x[:, 1]), :, 1)
+    return mapreduce(hcat, drop(eachcol(x), 1); init = init) do c
+        sb(c)
+    end
+end
+
+# Simplex adjoints
+
+function _simplex_bijector(X::TrackedVecOrMat, b::SimplexBijector)
+    return track(_simplex_bijector, X, b)
+end
+function _simplex_inv_bijector(Y::TrackedVecOrMat, b::SimplexBijector)
+    return track(_simplex_inv_bijector, Y, b)
+end
+
+@grad function _simplex_bijector(X::AbstractVector, b::SimplexBijector)
+    Xd = data(X)
+    return _simplex_bijector(Xd, b), Δ -> (simplex_link_jacobian(Xd)' * Δ, nothing)
+end
+@grad function _simplex_inv_bijector(Y::AbstractVector, b::SimplexBijector)
+    Yd = data(Y)
+    return _simplex_inv_bijector(Yd, b), Δ -> (simplex_invlink_jacobian(Yd)' * Δ, nothing)
+end
+
+@grad function _simplex_bijector(X::AbstractMatrix, b::SimplexBijector)
+    Xd = data(X)
+    return _simplex_bijector(Xd, b), Δ -> begin
+        mapreduce(hcat, eachcol(Xd), eachcol(Δ)) do c1, c2
+            simplex_link_jacobian(c1)' * c2
+        end, nothing
+    end
+end
+@grad function _simplex_inv_bijector(Y::AbstractMatrix, b::SimplexBijector)
+    Yd = data(Y)
+    return _simplex_inv_bijector(Yd, b), Δ -> begin
+        @views init = reshape(simplex_invlink_jacobian(Yd[:,1])' * Δ[:,1], :, 1)
+        mapreduce(hcat, drop(eachcol(Yd), 1), drop(eachcol(Δ), 1); init = init) do c1, c2
+            simplex_invlink_jacobian(c1)' * c2
+        end, nothing
+    end
+end
+
+logabsdetjac(b::SimplexBijector, x::TrackedVecOrMat) = track(logabsdetjac, b, x)
+@grad function logabsdetjac(b::SimplexBijector, x::AbstractVector)
+    xd = data(x)
+    return logabsdetjac(b, xd), Δ -> begin
+        (nothing, simplex_logabsdetjac_gradient(xd) * Δ)
+    end
+end
+@grad function logabsdetjac(b::SimplexBijector, x::AbstractMatrix)
+    xd = data(x)
+    return logabsdetjac(b, xd), Δ -> begin
+        @views init = reshape(simplex_logabsdetjac_gradient(xd[:,1]) * Δ[1], :, 1)
+        (nothing, mapreduce(hcat, drop(eachcol(xd), 1), drop(Δ, 1); init = init) do c, g
+            simplex_logabsdetjac_gradient(c) * g
+        end)
+    end
+end
+
+(ib::Inverse{<:Logit{<:TrackedReal}})(y::AbstractArray) = ib.(y)::TrackedArray
```

Comment:
Are these actually useful? Seems you'd still have unstability due to the element-type of `TrackedArray` being unknown, no?

Reply:
Hmm I thought I checked they helped, but now that I am double checking they don't seem to add much. I will remove them. Thanks!